contents/tutorials/redirect-testing.md

```diff
@@ -0,0 +1,398 @@
+
+To evaluate the experiment flag in our middleware, we need a distinct user ID to target. This needs to be consistent because we want the same user to see the same variant of the experiment every time.
+
+To do this, we need to:
+
+1. Check if a `distinct_id` exists in the PostHog cookie, and use it if so.
+2. Create a `distinct_id` if not.
+
+This requires using your project API key to get the cookies, parsing them as JSON, and potentially creating a distinct ID using `crypto.randomUUID()`. Altogether, this looks like this:
+
+```js
+import { NextResponse } from 'next/server'
+
+export async function middleware(request) {
+  const ph_project_api_key = '<ph_project_api_key>'
+  const ph_cookie_key = `ph_${ph_project_api_key}_posthog`
+  const cookie = request.cookies.get(ph_cookie_key);
+
+  let distinct_id;
+  if (cookie) {
+    // Use PostHog distinct_id 
+    distinct_id = JSON.parse(cookie.value).distinct_id;
+  } else {
+    // Create new distinct_id
+    distinct_id = crypto.randomUUID();
+  }
+//... rest of code
+```
+
+### Evaluating our redirect test with PostHog
+
+With our distinct ID, we use the PostHog API to check the value of the `main-redirect` feature flag for a user (because [we can’t use PostHog SDKs in Next.js middleware](https://vercel.com/docs/functions/edge-functions/edge-runtime#supported-apis)). This is known as evaluating the feature flag.
+
+Specifically, we evaluate the flag by making a POST request to the `https://app.posthog.com/decide?v=3` route (replace `app` with `eu` if you’re on EU Cloud) with your project API key and user distinct ID. From the response, we get the value of the `main-redirect` feature flag and use it to redirect to the right page. Altogether, it looks like this:
+
+```js
+// redirect-test/middleware.js
+//... rest of code
+
+	const requestOptions = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      api_key: ph_project_api_key,
+      distinct_id: distinct_id
+    })
+  };
+  
+  // Evaluate experiment flag
+  const ph_request = await fetch(
+    'https://app.posthog.com/decide?v=3', // or eu.posthog.com
+    requestOptions
+  );
+  const data = await ph_request.json();
+  const flagResponse = data.featureFlags['main-redirect']
+
+  // Redirect to correct page
+  if (request.nextUrl.pathname=== '/test' && flagResponse === 'control') {
+    return NextResponse.redirect(new URL('/control', request.url))
+  }
+  if (request.nextUrl.pathname=== '/control' && flagResponse === 'test') {
+    return NextResponse.redirect(new URL('/test', request.url))
+  }
+  return NextResponse.next()
+}
+//... rest of code
+```
+
+### Capturing exposure
+
+To get accurate results for our experiment, we also need to capture a `$feature_flag_called` event after the feature flag has been evaluated, but before redirecting. This is known as an exposure event and shows a user is part of the experiment. It requires another POST request like this:
```

Comment:
Think just an additional sentence is needed to clarify why we need to capture it.  Something along the lines of this.

Also, do we need the "but before redirecting" bit? I dont think so. But if we decide to include, we need to explain why
```suggestion
To get accurate results for our experiment, we also need to capture a `$feature_flag_called` event after the feature flag has been evaluated. This records which experiment variant each user was assigned to, and enables us to calculate results for the experiment.

This is known as an exposure event and shows a user is part of the experiment. It requires another POST request like this:
```